internal/dag/gatewayapi_processor.go

```diff
@@ -695,7 +695,7 @@ func (p *GatewayAPIProcessor) resolveListenerSecret(certificateRefs []gatewayapi
 			listenerName,
 			gatewayapi_v1beta1.ListenerConditionResolvedRefs,
 			metav1.ConditionFalse,
-			gatewayapi_v1beta1.ListenerReasonInvalidCertificateRef,
+			gatewayapi_v1beta1.ListenerReasonRefNotPermitted,
```

Comment:
Based on https://github.com/kubernetes-sigs/gateway-api/pull/1334 it looks to me like this one should have its reason stay as is -- i.e. `InvalidCertificateRef`.